conda_env/env.py

```diff
@@ -46,6 +46,14 @@ def from_environment(name, prefix):
     return Environment(name=name, dependencies=dependencies)
 
 
+def from_yaml(yamlstr, name=None):
+    """Load and return a ``Environment`` from a given ``yaml string``"""
+    data = yaml.load(yamlstr)
+    if name is not None:
+        data['name'] = name
+    return Environment(**data)
+
+
 def from_file(filename):
```

Comment:
```python
def from_file(filename):
    if not os.path.exists(filename):
        raise exceptions.EnvironmentFileNotFound(filename)
    with open(filename, "rb") as fp:
        return from_yaml(fp.read(), filename=filename)
```

Also, `from_yaml()` would need to be adjusted to take `kwargs`.  I think that works better than `name` as the only value you can provide explicitly.